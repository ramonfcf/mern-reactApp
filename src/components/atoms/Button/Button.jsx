import PropTypes from 'prop-types';


const Button = ({
  onClick,
  text,
  size = "large",
  color = "secondary",
  type = "normal",
}) => {
  const buttonSize = {
    small: "ms-2 btn-sm p-1",
    large: "",
  };

  const types = {
    normal: "btn text-white my-2 ",
    close: "btn-close",
    none: "",
  };

  const colors = {
    primary: "bg-primary",
    secondary: "bg-secondary",
    success: "bg-success",
    danger: "bg-danger",
    warning: "bg-warning",
    info: "bg-info",
    light: "bg-light",
    dark: "bg-dark",
    none: "",
  };

  return (
    <button
      type="button"
      className={`${types[type]} ${colors[color]} ${buttonSize[size]}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string,
  size: PropTypes.string,
  color: PropTypes.string,
  type: PropTypes.string,
};

