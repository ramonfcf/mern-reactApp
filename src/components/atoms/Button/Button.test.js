import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "./Button";

describe("Button component", () => {
  const mockOnClick = jest.fn();
  const buttonText = "Click me";

  it("matches snapshot", () => {
    const { container } = render(
      <Button
        onClick={mockOnClick}
        text={buttonText}
        type="normal"
        color="primary"
        size="large"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls onClick prop when clicked", () => {
    render(<Button onClick={mockOnClick} text={buttonText} />);
    const buttonElement = screen.getByRole("button");

    fireEvent.click(buttonElement);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it("renders with correct text", () => {
    render(<Button onClick={mockOnClick} text={buttonText} />);
    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toHaveTextContent(buttonText);
  });

  it("renders with correct size", () => {
    render(<Button onClick={mockOnClick} text={buttonText} size="small" />);
    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toHaveClass("btn-sm");
  });

  it("renders with correct color", () => {
    render(<Button onClick={mockOnClick} text={buttonText} color="primary" />);
    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toHaveClass("bg-primary");
  });
});
