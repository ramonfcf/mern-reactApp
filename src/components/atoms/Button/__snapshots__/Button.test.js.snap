// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<div>
  <button
    class="btn text-white my-2  bg-primary "
    type="button"
  >
    Click me
  </button>
</div>
`;
